src/language-js/comments/handle-comments.js

```diff
@@ -585,34 +585,6 @@ function handleLastFunctionArgComments({
     return true;
   }
 
-  if (
-    enclosingNode?.type === "FunctionDeclaration" &&
-    followingNode?.type === "BlockStatement"
-  ) {
-    const functionParamRightParenIndex = (() => {
-      const parameters = getFunctionParameters(enclosingNode);
-      if (parameters.length > 0) {
-        return getNextNonSpaceNonCommentCharacterIndex(
-          text,
-          locEnd(parameters.at(-1)),
-        );
-      }
-      const functionParamLeftParenIndex =
-        getNextNonSpaceNonCommentCharacterIndex(text, locEnd(enclosingNode.id));
-      return (
-        functionParamLeftParenIndex !== false &&
-        getNextNonSpaceNonCommentCharacterIndex(
-          text,
-          functionParamLeftParenIndex + 1,
-        )
-      );
-    })();
-    if (locStart(comment) > functionParamRightParenIndex) {
-      addBlockStatementFirstComment(followingNode, comment);
-      return true;
-    }
-  }
```

Comment:
This part is added in #5435, remove it doesn't break the case it fixes. [Playground](https://deploy-preview-15326--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEcAeAHCAnGACAMwFcowYBLaPIjAEwEN4BlGHOAMWwgFsBVAJQAyACgA6UPHgDOrbHDwB6AFRI8LNkoUAacZOBS49bGAAWWvCfpSTAX0Uq8+w8ZOqZ2clADm5y9bcwHt42muIAlI42IFogEBgU0FLIoEZcAO4ACkYISSj0ADZp9ACeSTEARtj0YADWcDBM9Nxwgp5wyAQFBhVVtfVMGNWeXsiBRHAxBtzko9jjk8P5cACKRBDwHV0TIABWUmhMiytrG0id+d0gAI4ncBlcGLkgVgC0UHBwtJ-RIIH05PlhgBhHjcejIZ75fI-KRHACCMEC5HKRHgGTg2Fa702F22Jhg3HyAHUTOR4FJBmA4EwcmTyAA3MnFCFgKRlED08YASSgX1gTDAHnicN5LGKSxxlwwXAMRKqGAh0rgBmw9PaMU8Kpg93oXjBku2g2wKoh5Xo5Tg0Ji0s8MCJ5FoMBMyAAHAAGGJyG7kOQ6vXgs5bGIwc32x3OpAAJhiRAMABVzblzpc4NwLbQvrRBPRvERdRwcGDEcMIfRURAQDYbEA)

Reply:
But breaks #5250 ...